File: 코드리뷰/한종호(lgwl)/8일차/코드리뷰_바이러스.cpp
#include <iostream>
using namespace std;

int N, M, ans = 0;
int map[101][101], visited[101];

void dfs(int idx)
{
	visited[idx] = 1;
	ans++;
	for (int i = 1; i <= N; i++)
		if (!visited[i] && map[idx][i])
			dfs(i);
}

int main()
{
	ios::sync_with_stdio(0);
	cin.tie(0); cout.tie(0);

	cin >> N >> M;
	for (int i = 0; i < M; i++)
	{
		int a, b;
		cin >> a >> b;
		map[a][b] = map[b][a] = 1;
	}

	dfs(1);
	cout << ans - 1 << "\n";

	return (0);
}

File: 코드리뷰/김정혜(tweetiijh)/8일차/코드리뷰_바이러스.py
n=int(input())
k=int(input()) # 쌍의 수
visited=[]

for i in range(n+1):
    visited.append(0)
graph=[[0] *(n+1) for i in range(n+1)]

for i in range(k):
    x,y=map(int,input().split())
    graph[x][y]=1
    graph[y][x]=1
    

def dfs(x):
    visited[x]=1
    for i in range(1,n+1):
        if visited[i]==0 and graph[x][i]==1:
            
            dfs(i)

dfs(1)
ans=0
for i in range(2,n+1):
    if visited[i]!=0:
        ans+=1
        


        
print(ans)


    
